neqo-transport/src/connection.rs

```diff
@@ -3694,4 +3734,181 @@ mod tests {
         let c_tx_dgrams = send_bytes(&mut client, 0, now);
         assert_eq!(c_tx_dgrams.len(), 4);
     }
+
+    fn check_is_packet_drop(peer: &mut Connection, pkt: Option<Datagram>, dropped: u64, dups: u64) {
+        let dropped_before = peer.stats.dropped_rx;
+        let dups_before = peer.stats.dups_rx;
+        let out = peer.process(pkt, now());
+        assert!(out.as_dgram_ref().is_none());
+        assert_eq!(dropped, peer.stats.dropped_rx - dropped_before);
+        assert_eq!(dups, peer.stats.dups_rx - dups_before);
+    }
+
+    #[test]
+    fn test_discarded_initial_keys() {
+        qdebug!("---- client: generate CH");
+        let mut client = default_client();
+        let init_pkt_c = client.process(None, now());
+        assert!(init_pkt_c.as_dgram_ref().is_some());
```

Comment:
Try this instead:

```rust
let init_pkt_c = client.process(None, now()).dgram();
assert!(init_pkt_c.is_some());
```
It's much easier to work with in general.  Less use of `as_dgram_ref()` and `unwrap`.

The line below would be regular rust:
```rust
assert_eq!(init_pkt_c.as_ref().unwrap().len(), 1232);
```

And if you wanted to clone the datagram, it is the much simpler:

```rust
let init_pkt_s = server.process(init_pkt_c.clone(), now()); 
```